### ISSUE TITLE
Cann`t use/ Help me 

### ISSUE THREAD

**Issue by ghost** [GMS-SA/roundcube-advanced-search#31]
#5 {main}
  thrown in D:\Test\plugins\advanced_search\advanced_search.php on line 561
[08-Feb-2022 01:00:26 Europe/Minsk] PHP Fatal error:  Uncaught Error: Call to undefined method rcmail::build_folder_tree() in D:\Test\plugins\advanced_search\advanced_search.php:561
Stack trace:
#0 D:\Test\plugins\advanced_search\advanced_search.php(445): advanced_search->folder_selector(Array)
#1 D:\Test\plugins\advanced_search\advanced_search.php(422): advanced_search->generate_searchbox()
#2 D:\Test\program\lib\Roundcube\rcube_plugin_api.php(570): advanced_search->display_advanced_search()
#3 D:\Test\program\include\rcmail.php(248): rcube_plugin_api->exec_action('plugin.display_...')
#4 D:\Test\index.php(283): rcmail->action_handler()
#5 {main}
  thrown in D:\Test\plugins\advanced_search\advanced_search.php on line 561
how Fix it? Thanks. 
Roundcube Version 1.5.2

**Comment by jab4** [comment]
Hi,
Without further knowing much details of the roundcube inner workings, this looks like a namespace issue. 
```
Call to undefined method rcmail::build_folder_tree
```
...is a static function call to the class named `rcmail`, which does exist in `#4: rcmail->action_handler`, otherwise the steps before that could have never taken place. (EDIT: PHP stack traces are actually backwards, because `#5` is the entry point and `#0` is the last point run that triggered the error.)

In other words: `advanced_search` apparently has a `namespace _whatever_` line somewhere, which breaks structural reworking that were introduced in Roundcube 1.5 (or even earlier).

Is this a built-in plugin? I doubt that, others would have this issue and you wouldn't wait here for an answer. Where did you get the plugin? When was it last updated? Could you post a link?

**Comment by jab4** [comment]
Oh my. Please forget my previous request for the link. Captain Obvious just said hello. How did I end up here? 🤔

**Comment by Dmitry422** [comment]
I have the same error with Roundcube 1.6
I hope plugin author read this thread and help us )

Thank you!

**Comment by c---** [comment]
@Dmitry422 Try this fork of a fork: https://github.com/mxroute/advanced_search
